hack/generator/pkg/codegen/pipeline_mark_storage_version.go

```diff
@@ -0,0 +1,112 @@
+/*
+ * Copyright (c) Microsoft Corporation.
+ * Licensed under the MIT license.
+ */
+
+package codegen
+
+import (
+	"context"
+	"sort"
+
+	"github.com/Azure/k8s-infra/hack/generator/pkg/astmodel"
+	"github.com/pkg/errors"
+)
+
+// exportPackages creates a PipelineStage to export our generated code as a set of packages
+func markStorageVersion() PipelineStage {
+	return MakePipelineStage(
+		"markStorageVersion",
+		"Marking the latest version of type as the storage version",
+		func(ctx context.Context, types astmodel.Types) (astmodel.Types, error) {
+			updatedDefs, err := MarkLatestResourceVersionsForStorage(types)
+			if err != nil {
+				return nil, errors.Wrapf(err, "unable to mark latest resource versions for as storage versions")
+			}
+
+			return updatedDefs, nil
+		})
+}
+
+// MarkLatestResourceVersionsForStorage marks the latest version of each resource as the storage version
+func MarkLatestResourceVersionsForStorage(types astmodel.Types) (astmodel.Types, error) {
+
+	result := make(astmodel.Types)
+	resourceLookup, err := groupResourcesByVersion(types)
+	if err != nil {
+		return nil, err
+	}
+
+	for _, def := range types {
+		// see if it is a resource
+		if resourceType, ok := def.Type().(*astmodel.ResourceType); ok {
```

Comment:
Need to restrict the loop to looking in Storage packages - use `IsStoragePackageReference`

Reply:
My intent here was just to move the existing logic that was in `pipeline_export_generated_code` to its own stage, not necessarily to change the logic of it.

Did you want me to change the logic too?

Reply:
Happy for this to stay as it is - I can tweak as a part of upcoming PRs.